lib/matplotlib/axis.py

```diff
@@ -1643,6 +1643,19 @@ def set_ticklabels(self, ticklabels, *args, minor=False, **kwargs):
         # replace the ticklabels list with the processed one
         ticklabels = get_labels
 
+        get_labels = []
+        if(rcParams['text.usetex'] and
+            len(rcParams['text.latex.preamble']) != 0):
+            for presemble in rcParams['text.latex.preamble']:
+                if('usepackage{'  in presemble and 'amsmath' in presemble):
+                    # check text.usetex and amsmath package 
+                    # if True use amsmath package for the negative sign
+                    for t in ticklabels:
+                        # replace latex "-" sign with latex.amsmath "-" sign
+                        get_labels.append(t.replace('-','\\text{-}'))
```

Comment:
Would mathrm work?

Reply:
`\mathrm{-} `does not work. I tried. But maybe `\operatorname{-}`. In that case **presemble** check won't be required.

Reply:
\mathrm{-}does not work. I tried. But maybe \operatorname{-}. In that case presemble check won't be required.